paper/paper.md

```diff
@@ -0,0 +1,84 @@
+
+In recent years the IPython project introduced the *Jupyter notebook* [@kluyver2016jupyter], an interface that is particularly conducive to interactive and literate computing where programmers can interleave prose with code and figures. The Jupyter notebook is ideal for educators because it allows them to create assignments which include instructions along with cells, in which students can provide solutions to exercises. Students can, for example, be asked to write code both to compute and visualize a particular result. Because of the interactive nature of the notebook, students can iterate on a coding problem without having to switch back and forth between the command line and a text editor, and they can see the results of their code almost instantly.
+
+Instructors in many fields have already begun using the Jupyter notebook as a teaching platform. The notebook has appeared in over 70 classes [@Castano_Jupyter_Map_Dataset] on subjects including geology, mathematics, mechanical engineering, data science, chemical engineering, and bioinformatics, just to name a few. Software Carpentry, which aims to teach graduate students basic computational skills, has also adopted the notebook for some of its lessons [@Wilson2014].
+
+Despite its appearance in many classrooms—prior to the existence of nbgrader—the notebook was rarely used on a large scale for *graded* assignments. Instead, it was often used either for ungraded in-class exercises; or in classes small enough that notebooks can be graded by hand. This is because there are several challenges to using the notebook for graded assignments at scale. First, for large class sizes, it is not feasible for an instructor to manually grade the code that students write: there must be a way of autograding the assignments. A Jupyter notebook is not a typical script that can be run and may contain multiple parts of a problem within the same notebook. Second, for many courses, the programming is a means to an end for understanding concepts in a specific domain. Instructors may also want students to provide written free-responses interpreting the results of their code, and thus need to be able to rely on autograding for the coding parts of an assignment, but also be able to manually grade the written responses in the surrounding context of the student's code. Third, the process of distributing assignments to students and later collecting them can be tedious, and this is true even more so with Jupyter notebooks because there is a separate interface for accessing them beyond the standard system file browser. This often leads to confusion on the part of students about how to open notebooks after downloading them, and where to find the notebooks in order to submit them.
```

Comment:
- [ ] => `Despite its appearance in many classrooms, before the creation of nbgrader, the notebook was rarely used...`
- [ ] `;` to `,` in the second sentence.
- [ ] `can` to `could` in second sentence (tense issue).
- [ ] `large class sizes` => `large classes` (sizes is redundant when using large)
- [ ] `there must be a way of autograding the assignments` => `there must be a way to autograde the assignments`
- [ ] This sentence reads like a non sequitur: "A Jupyter notebook is not a typical script that can be run and may contain multiple parts of a problem within the same notebook." What is its relation to the previous or following sentence?
- [ ] "the programming" => "programming" to avoid a slight miscue ("what programming?", i asked)
- [ ] => "programming is a means to an end: understanding concepts in a scientific domain."
- [ ] Is the "their" in "their" code the students or the instructors?
- [ ] Hm, maybe it's actually this sentence that is out of place: "Second, for many courses, the programming is a means to an end for understanding concepts in a specific domain." When it's removed, everything makes more sense.
- [ ] ", and this is true even more so with" => ", even more so with"